File: Java/Array/Merge Sorted Array.java
//https://leetcode.com/problems/merge-sorted-array/
class Solution {
    public void merge(int[] num1, int m, int[] num2, int n) {
        int[] num3 = new int[m + n];
        int i = 0, j = 0, k = 0;

        while (i < m && j < n) {
            if (num1[i] < num2[j]) {
                num3[k++] = num1[i++];
            } else {
                num3[k++] = num2[j++];
            }
        }

        while (i < m) {
            num3[k++] = num1[i++];
        }

        while (j < n) {
            num3[k++] = num2[j++];
        }

        for (int x = 0; x < m + n; x++) {
            num1[x] = num3[x];
        }
    }
}

File: Python/Array/Merge Sorted Array.py
//https://leetcode.com/problems/merge-sorted-array/
class Solution(object):
    def merge(self, nums1, m, nums2, n):
        """
        :type nums1: List[int]
        :type m: int
        :type nums2: List[int]
        :type n: int
        :rtype: None Do not return anything, modify nums1 in-place instead.
        """
        num3 = []
        i, j = 0, 0
        
        # Merge the two sorted lists
        while i < m and j < n:
            if nums1[i] < nums2[j]:
                num3.append(nums1[i])
                i += 1
            else:
                num3.append(nums2[j])
                j += 1
        
        # Add remaining elements from nums1
        while i < m:
            num3.append(nums1[i])
            i += 1
        
        # Add remaining elements from nums2
        while j < n:
            num3.append(nums2[j])
            j += 1
        
        # Modify nums1 in-place
        for i in range(m + n):
            nums1[i] = num3[i]
